lib/puppet/settings/integer_setting.rb

```diff
@@ -0,0 +1,16 @@
+class Puppet::Settings::IntegerSetting < Puppet::Settings::BaseSetting
+  def munge(value)
+    return value if Integer === value
+    if value && value.match?(/\A[-+]?\d+\z/)
+      value = value.to_i
```

Comment:
Maybe just call ‘Integer(value)’ and rescue Argument error if its not already an instance of an integer or can’t be converted?